support-modules/aws/src/main/scala/com/gu/aws/AwsCloudWatchMetricPut.scala

```diff
@@ -27,32 +29,40 @@ object AwsCloudWatchMetricPut {
   )
 
   def apply(client: CloudWatchClient)(request: MetricRequest): Try[Unit] = {
+    logger.info("Logging metric: " + request)
 
-    val metricDatum1 = request.dimensions
-      .foldLeft(
-        MetricDatum
+    val javaDimensions = request.dimensions
+      .map { case (name, value) =>
+        Dimension
           .builder()
-          .metricName(request.name.value),
-      ) { case (agg, (name, value)) =>
-        agg.dimensions(
-          Dimension
-            .builder()
-            .name(name.value)
-            .value(value.value)
-            .build(),
-        )
+          .name(name.value)
+          .value(value.value)
+          .build()
       }
-      .value(request.value)
-      .unit(StandardUnit.COUNT)
-      .build()
+      .toSeq
+      .asJava
+
+    val metricDatum1 =
+      MetricDatum
+        .builder()
+        .metricName(request.name.value)
+        .dimensions(javaDimensions)
+        .value(request.value)
+        .unit(StandardUnit.COUNT)
+        .build()
 
     val putMetricDataRequest = PutMetricDataRequest
       .builder()
       .namespace(request.namespace.value)
       .metricData(metricDatum1)
       .build()
 
-    Try(client.putMetricData(putMetricDataRequest)).map(_ => ())
+    val attempt = Try(client.putMetricData(putMetricDataRequest))
+    attempt match {
+      case Failure(exception) => logger.warn("metric send failed with " + exception.toString)
```

Comment:
```suggestion
      case Failure(exception) => logger.error("metric send failed with " + exception.toString)
```

?

Reply:
yeah makes sense, I was assuming someone further up the tree[1] would log it properly and I wanted to avoid doubling up, but I don't see the harm.
[1]
https://github.com/guardian/support-frontend/blob/22d14edf4db1e4b5a4be7fc5c1c22369af16be30/support-workers/src/main/scala/com/gu/support/workers/lambdas/SendAcquisitionEvent.scala#L44